examples/js/loaders/GLTFLoader.js

```diff
@@ -1153,103 +1153,121 @@ THREE.GLTFLoader = ( function () {
+		if ( geometry.attributes.position === undefined ) hasMorphPosition = false;
+		if ( geometry.attributes.normal === undefined ) hasMorphNormal = false;
+
+		if ( ! hasMorphPosition && ! hasMorphNormal ) return;
+
+		var morphPositions = [];
+		var morphNormals = [];
 
 		for ( var i = 0, il = targets.length; i < il; i ++ ) {
 
 			var target = targets[ i ];
 			var attributeName = 'morphTarget' + i;
 
-			var positionAttribute, normalAttribute;
+			if ( hasMorphPosition ) {
 
-			if ( target.POSITION !== undefined ) {
+				var positionAttribute;
 
-				// Three.js morph formula is
-				//   position
-				//     + weight0 * ( morphTarget0 - position )
-				//     + weight1 * ( morphTarget1 - position )
-				//     ...
-				// while the glTF one is
-				//   position
-				//     + weight0 * morphTarget0
-				//     + weight1 * morphTarget1
-				//     ...
-				// then adding position to morphTarget.
-				// So morphTarget value will depend on mesh's position, then cloning attribute
-				// for the case if attribute is shared among two or more meshes.
+				if ( target.POSITION !== undefined ) {
 
-				positionAttribute = cloneBufferAttribute( accessors[ target.POSITION ] );
-				var position = geometry.attributes.position;
+					// Three.js morph formula is
+					//   position
+					//     + weight0 * ( morphTarget0 - position )
+					//     + weight1 * ( morphTarget1 - position )
+					//     ...
+					// while the glTF one is
+					//   position
+					//     + weight0 * morphTarget0
+					//     + weight1 * morphTarget1
+					//     ...
+					// then adding position to morphTarget.
+					// So morphTarget value will depend on mesh's position, then cloning attribute
+					// for the case if attribute is shared among two or more meshes.
 
-				for ( var j = 0, jl = positionAttribute.count; j < jl; j ++ ) {
+					positionAttribute = cloneBufferAttribute( accessors[ target.POSITION ] );
+					var position = geometry.attributes.position;
 
-					positionAttribute.setXYZ(
-						j,
-						positionAttribute.getX( j ) + position.getX( j ),
-						positionAttribute.getY( j ) + position.getY( j ),
-						positionAttribute.getZ( j ) + position.getZ( j )
-					);
+					for ( var j = 0, jl = positionAttribute.count; j < jl; j ++ ) {
 
-				}
+						positionAttribute.setXYZ(
+							j,
+							positionAttribute.getX( j ) + position.getX( j ),
+							positionAttribute.getY( j ) + position.getY( j ),
+							positionAttribute.getZ( j ) + position.getZ( j )
+						);
 
-			} else if ( geometry.attributes.position ) {
+					}
 
-				// Copying the original position not to affect the final position.
-				// See the formula above.
-				positionAttribute = cloneBufferAttribute( geometry.attributes.position );
+				} else {
 
-			}
+					// Copying the original position not to affect the final position.
+					// See the formula above.
+					positionAttribute = cloneBufferAttribute( geometry.attributes.position );
```

Comment:
Trying to remember why this case is here... is the problem that because `morphAttributes` takes the form:

```
morphAttributes = {
  position: [ position1, position2 ],
  normal: [ normal1, normal2 ]
}
```

... then if any morph target includes `normal`, every other target must also do so to avoid throwing off the array indices? And if so, what would "affect the final position" so that we need to make a copy of the buffer?

Reply:
Three.js morphPosition/Normal values are absolute. The formula is

    finalPosition/Normal =
        basePosition/Normal
            + weight0 * ( morphPosition/Normal0 - basePosition/Normal )
            + weight1 * ( morphPosition/Normal1 - basePosition/Normal )
            ...

while the glTF ones are relative

    finalPosition/Normal =
        basePosition/Normal
            + weight0 * glTFmorphPosition/Normal0
            + weight1 * glTFmorphPosition/Normal1
            ...

If morphPosition/Normal values in Three.js are same as the base ones, they don't affect finalPosition/Normal.

Imagine the following `targets` in glTF. 

    "targets": [
        {
            "NORMAL": 0
        },
        {
            "POSITION": 1,
            "NORMAL": 2
        }
    ]

targets[0] doesn't have "POSITION" so `mesh.morphTargetInfluences[0]` shouldn't affect finalPosition. Then I need to copy base position (`geometry.attributes.position`) to `geometry.morphAttributes.position[0]`.

Reply:
Ok yes, that all makes sense except one thing — do we need to clone the BufferAttribute if we're not going to modify it? Why not just reuse the existing BufferAttribute? In this example, I think it would be OK to have `geometry.attributes.position === geometry.morphAttributes.position[0]`. If there's a reason it must be cloned let's just add a comment.

Reply:
I had just never thought and tried to assign the same array to two or more attributes in shader. I tried and it seems to work. So updated. Thanks.